### PR TITLE
Use `Writable` utility type to improve typings

### PR DESCRIPTION
**Pull request by JeanMeche**
The `Writable` type is usefull when we want overwrite readonly properties and we still want to maintain code navigation/reference. It should be use instead of `any` or `{myPrivateKey: any}` type assertions for example.

See individual commits

## PR Type
What kind of change does this PR introduce?

- [x] Refactoring (no functional changes, no api changes)

## Does this PR introduce a breaking change?

- [x] No

### CODE REVIEW

File: packages/forms/src/directives/ng_control_status.ts [hunk 1]
```diff
@@ -54,7 +57,7 @@ export class AbstractControlStatus {
   protected get isSubmitted() {
     // We check for the `submitted` field from `NgForm` and `FormGroupDirective` classes, but
     // we avoid instanceof checks to prevent non-tree-shakable references to those types.
-    return !!(this._cd as unknown as {submitted: boolean} | null)?.submitted;
+    return !!(this._cd as Writable<NgForm|FormGroupDirective>| null)?.submitted;
```
Comment (JeanMeche) [inline packages/forms/src/directives/ng_control_status.ts:60]: This should break the treeshaking right ?

File: packages/core/src/linker/query_list.ts [hunk 2]
```diff
@@ -154,16 +155,13 @@ export class QueryList<T> implements Iterable<T> {
    *    are compared as is (without any pre-processing).
    */
   reset(resultsTree: Array<T|any[]>, identityAccessor?: (value: T) => unknown): void {
-    // Cast to `QueryListInternal` so that we can mutate fields which are readonly for the usage of
-    // QueryList (but not for QueryList itself.)
-    const self = this as QueryListInternal<T>;
-    (self as {dirty: boolean}).dirty = false;
+    (this as {dirty: boolean}).dirty = false;
```
Comment (JeanMeche) [inline packages/core/src/linker/query_list.ts:158]: Any idea why this property is of type `true` and not a boolean ?

File: packages/forms/src/directives/ng_control_status.ts (hunk 1 above)
Comment (garrettld) [reply packages/forms/src/directives/ng_control_status.ts:59]: You could use type-only imports to be certain that it won't, e.g. `import { type NgForm } from './ng_form'`.

Comment (JeanMeche) [reply packages/forms/src/directives/ng_control_status.ts:59]: Great idea, thanks !

**Comment by dylhunn** [comment]
This is a nice change conceptually. I just need to sit down and look over all the usage points.

**Comment by JeanMeche** [comment]
@dylhunn Should be really straight forward 😊

**Comment by JeanMeche** [comment]
The presubmit is failing, any info on what is breaking ?

**Review by jessicajaniuk — APPROVED** [review]
reviewed-for: fw-animations

**Review by dylhunn — APPROVED** [review]
reviewed-for: fw-core, fw-forms

**Comment by dylhunn** [comment]
g3 presubmit is passing.

**Comment by dylhunn** [comment]
disabling pullapprove -- router change is a no-op

**Comment by dylhunn** [comment]
This PR was merged into the repository by commit e5d327d0496d23578cf09b548f7c3a14774407fc.

**Pull request closed without merging**